EmmyLua/SharedXML/SharedColorConstants.lua

```diff
@@ -0,0 +1,369 @@
+
+---@type ColorMixin
+ORANGE_FONT_COLOR = {}
+
+---@type ColorMixin
+PASSIVE_SPELL_FONT_COLOR = {}
+
+---@type ColorMixin
+BATTLENET_FONT_COLOR = {}
+
+---@type ColorMixin
+TRANSMOGRIFY_FONT_COLOR = {}
+
+---@type ColorMixin
+DISABLED_FONT_COLOR = {}
+
+---@type ColorMixin
+BRIGHTBLUE_FONT_COLOR = {}
+
+---@type ColorMixin
+LIGHTBLUE_FONT_COLOR = {}
+
+---@type ColorMixin
+LIGHTGRAY_FONT_COLOR = {}
+
+---@type ColorMixin
+PAPER_FRAME_EXPANDED_COLOR = {}
+
+---@type ColorMixin
+PAPER_FRAME_COLLAPSED_COLOR = {}
+
+---@type ColorMixin
+PAPER_FRAME_DARK_COLOR = {}
+
+---@type ColorMixin
+PAPER_FRAME_TITLE_COLOR = {}
+
+---@type ColorMixin
+PAPER_FRAME_TEXT_COLOR = {}
+
+---@type ColorMixin
+INVALID_EQUIPMENT_COLOR = {}
+
+
+
+---@type ColorMixin
+ARTIFACT_BAR_COLOR = {}
+
+---@type ColorMixin
+WARBOARD_OPTION_TEXT_COLOR = {}
+
+---@type ColorMixin
+DEFAULT_CHAT_CHANNEL_COLOR = {}
+
+---@type ColorMixin
+DIM_GREEN_FONT_COLOR = {}
+
+---@type ColorMixin
+BLACK_FONT_COLOR = {}
+
+---@type ColorMixin
+LINK_FONT_COLOR = {}
+
+---@type ColorMixin
+SEPIA_COLOR = {}
+
+---@type ColorMixin
+HIGHLIGHT_LIGHT_BLUE = {}
+
+---@type ColorMixin
+CORRUPTION_COLOR = {}
+
+---@type ColorMixin
+LORE_TEXT_BODY_COLOR = {}
+
+---@type ColorMixin
+RARE_MISSION_COLOR = {}
+
+---@type ColorMixin
+TUTORIAL_FONT_COLOR = {}
+
+---@type ColorMixin
+DARKGRAY_COLOR = {}
+
+---@type ColorMixin
+SCENARIO_STAGE_COLOR = {}
+
+---@type ColorMixin
+TRADESKILL_EXPERIENCE_COLOR = {}
+
+---@type ColorMixin
+SUBSCRIPTION_INTERSTITIAL_COLOR = {}
+
+---@type ColorMixin
+GLUE_DIALOG_FONT_COLOR = {}
+
+
+
+---@type ColorMixin
+QUEST_OBJECTIVE_FONT_COLOR = {}
+
+---@type ColorMixin
+QUEST_OBJECTIVE_HIGHLIGHT_FONT_COLOR = {}
+
+---@type ColorMixin
+QUEST_OBJECTIVE_DISABLED_FONT_COLOR = {}
+
+---@type ColorMixin
+QUEST_OBJECTIVE_DISABLED_HIGHLIGHT_FONT_COLOR = {}
+
+
+
+---@type ColorMixin
+AREA_NAME_FONT_COLOR = {}
+
+---@type ColorMixin
+AREA_DESCRIPTION_FONT_COLOR = {}
+
+---@type ColorMixin
+INVASION_FONT_COLOR = {}
+
+---@type ColorMixin
+INVASION_DESCRIPTION_FONT_COLOR = {}
+
+
+
+---@type number
+local n = 0
+
+---@type table<number, table<string, number>>
+FACTION_BAR_COLORS = {
+	[1] = {r = n, g = n, b = n},
+	[2] = {r = n, g = n, b = n},
+	[3] = {r = n, g = n, b = n},
+	[4] = {r = n, g = n, b = n},
+	[5] = {r = n, g = n, b = n},
+	[6] = {r = n, g = n, b = n},
+	[7] = {r = n, g = n, b = n},
+	[8] = {r = n, g = n, b = n},
+}
+
+
+
+---@type table<string, number[]>
+MATERIAL_TEXT_COLOR_TABLE = {
+	["Default"] = {},
+	["Stone"] = {},
+	["Parchment"] = {},
+	["Marble"] = {},
+	["Silver"] = {},
+	["Bronze"] = {},
+	["ParchmentLarge"] = {}
+}
+
+
+
+---@type table<string, number[]>
+MATERIAL_TITLETEXT_COLOR_TABLE = {
+	["Default"] = {},
+	["Stone"] = {},
+	["Parchment"] = {},
+	["Marble"] = {},
+	["Silver"] = {},
+	["Bronze"] = {},
+	["ParchmentLarge"] = {}
+}
+
+
+
+---@type ColorMixin
+FRIENDS_BNET_NAME_COLOR = {}
+
+---@type ColorMixin
+FRIENDS_BNET_BACKGROUND_COLOR = {}
```

Comment:
For `---@type` I wouldn't mind if we made it more compact like this
```lua
FRIENDS_BNET_NAME_COLOR = {} ---@type ColorMixin
FRIENDS_BNET_BACKGROUND_COLOR = {} ---@type ColorMixin
```

Reply:
I can remove blank lines, whitespaces. But writing annotations to the side is bad practice (it's easy to get confused) and inconvenient

Reply:
I generally agree it's a bad practice, it can mess up things and confuse the language server. Just for simple variables like this I'd make an exception and put it to the side, to be honest.